### ISSUE TITLE
no match if `+` after pattern

### ISSUE THREAD

**Issue by jonorthwash** [apertium/apertium-separable#13]
Acceptable translation, with good match:
```
$ echo "бүгүнкү күндө түшөт" | apertium -d . kir-eng
he falls these days

$ echo "бүгүнкү күндө түшөт" | apertium -d . kir-eng-disam  # and some manual cleaning
^бүгүн<adv><attr>$ ^күн<n><loc>$ ^түш<v><iv><aor><p3><sg>$^.<sent>$

$ echo "бүгүнкү күндө түшөт" | apertium -d . kir-eng-autoseq
^бүгүнкү күндө<adv>$ ^түш<v><iv><aor><p3><sg>$^.<sent>$

$ echo "бүгүнкү күндө түшөт" | apertium -d . kir-eng-biltrans
^бүгүнкү күндө<adv>/these days<adv>/today<adv>$ ^түш<v><iv><aor><p3><sg>/fall<vblex><aor><p3><sg>$^.<sent>/.<sent>$
```

Problematic translation:
```
$ echo "бүгүнкү күндө" | apertium -d . kir-eng
he is on the today day

$ echo "бүгүнкү күндө" | apertium -d . kir-eng-disam # and some manual cleaning
^бүгүн<adv><attr>$ ^күн<n><loc>+э<cop><aor><p3><sg>$^.<sent>$

$ echo "бүгүнкү күндө" | apertium -d . kir-eng-autoseq
^бүгүн<adv><attr>$ ^күн<n><loc>+э<cop><aor><p3><sg>$^.<sent>$

$ echo "бүгүнкү күндө" | apertium -d . kir-eng-biltrans
^бүгүн<adv><attr>/today<adv><attr>$ ^күн<n><loc>/day<n><loc>/sun<n><loc>$ ^э<cop><aor><p3><sg>/be<vbser><aor><p3><sg>$^.<sent>/.<sent>$
```

The reason the second translation is problematic is because the current rule doesn't match because of the `+э<cop>...`.  Here's the current rule:

```
		<e lm="бүгүнкү күндө" c="today">
			<p>
				<l>бүгүн<s n="adv"/><s n="attr"/><j/>күн<s n="n"/><s n="loc"/></l>
				<r>бүгүнкү<b/>күндө<s n="adv"/></r>
			</p>
			<i><j/></i>
		</e>
```

**Comment by mr-martian** [comment]
You could fix this by moving pretransfer before separable, then there wouldn't be any `+`s.

**Comment by jonorthwash** [comment]
Hmm...  I think sometimes we'll want to refer to `+` in apertium-separable?

**Comment by jonorthwash** [comment]
In one pair, I have `+` that I need to refer to.

I think the best solution here is to have a symbol stand for `+`, and perhaps a symbol that can be "space or `+`".  @ftyers, what do you think?  (Suggestions for symbols?)

**Comment by mr-martian** [comment]
@jonorthwash I think this is what you want:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<dictionary type="sequential">
  <alphabet>abcdefghijklmnopqrstuvwxyz</alphabet>
  <sdefs>
    <sdef n="n"/>
  </sdefs>

  <pardefs>
    <pardef n="maybe_comp">
      <e>
        <p><l>+</l><r><j/></r></p>
        <i><w/><t/><j/></i>
      </e>
      <e><i><j/></i></e>
    </pardef>
  </pardefs>

  <section id="main" type="standard">
    <e>
      <p><l>x<s n="n"/>+y<s n="n"/></l><r>x<b/>y<s n="n"/></r></p>
      <i><t/></i>
      <par n="maybe_comp"/>
    </e>
  </section>
</dictionary>
```
```bash
$ lsx-comp lr test.dix test.bin
main@standard 11 15
$ echo "^x<n>+y<n><sg>$" | lsx-proc test.bin 
^x y<n><sg>$
$ echo "^x<n>+y<n><sg>+z<adj>$" | lsx-proc test.bin 
^x y<n><sg>$ ^z<adj>$
```

**Comment by ftyers** [comment]
Well, `<j/>` is traditionally `+` but we're using it for between words, you can use a literal `+` but it's a bit ugly in XML. We could use `<j/>` for `+` and then some letter from `b c d f h j k m n o q r u v w x y z` for the word splitter in lsx.

**Comment by jonorthwash** [comment]
> Well, `<j/>` is traditionally `+` but we're using it for between words, you can use a literal `+` but it's a bit ugly in XML. We could use `<j/>` for `+` and then some letter from `b c d f h k m n o q r u v w x y z` for the word splitter in lsx.

That would require updating a whole bunch of lsx files across the codebase?  But I guess we're planning on language-specific (instead of pair-specific) lsx files anyway, so maybe it could all be done in one fell swoop?

**Comment by jonorthwash** [comment]
> @jonorthwash I think this is what you want:

Yep, that seems to work!
https://github.com/apertium/apertium-eng-kir/commit/f513c18ca2689105b9dae6d3c75fea18655b3f9e

**Comment by jonorthwash** [comment]
I'm going to have to do something similar for modal clitics, I think.  It's not ideal, but it works.

Should we continue to consider a solution for a symbol that is "any word edge", whether a space or +?  I think it would allow things to be streamlined a bit.

**Comment by ftyers** [comment]
@jonorthwash I don't think so because sometimes we need to output +, like in my váyanse example.

**Comment by ftyers** [comment]
> > Well, `<j/>` is traditionally `+` but we're using it for between words, you can use a literal `+` but it's a bit ugly in XML. We could use `<j/>` for `+` and then some letter from `b c d f h k m n o q r u v w x y z` for the word splitter in lsx.
> 
> That would require updating a whole bunch of lsx files across the codebase? But I guess we're planning on language-specific (instead of pair-specific) lsx files anyway, so maybe it could all be done in one fell swoop?

I don't think it's a big problem and could be scripted easily.

**Comment by jonorthwash** [comment]
> > Should we continue to consider a solution for a symbol that is "any word edge", whether a space or +? I think it would allow things to be streamlined a bit.
>
> @jonorthwash I don't think so because sometimes we need to output +, like in my váyanse example.

I don't see how adding a new symbol makes this a problem.  Could you explain?

**Comment by jonorthwash** [comment]
> > @jonorthwash I think this is what you want:
> 
> Yep, that seems to work!
> [apertium/apertium-eng-kir@f513c18](https://github.com/apertium/apertium-eng-kir/commit/f513c18ca2689105b9dae6d3c75fea18655b3f9e)

This is actually a better solution: https://github.com/apertium/apertium-eng-kir/commit/c1dd7c54f2922f9d2afbdf3536d37dcd41350a45

The problem is, I'm going to use `w_edge` all over the place now.  That's what I'd like to create a symbol for.

**Comment by mr-martian** [comment]
If you want it to be possible for `<d/>` to match `+` or `$` and output the same, it wouldn't be that hard to do that (have an attribute on `<d/>` that inserts `(+(<ANY_TAG>|<ANY_CHAR>)*)?`). Perhaps `<d cmp="yes"/>`?

On the other hand, if you want there to be cases where you read in `+` and output `$`, that will have to be manual.
